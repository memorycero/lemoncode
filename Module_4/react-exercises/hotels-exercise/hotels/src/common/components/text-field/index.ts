export * from "../../components/text-field/text-field-form.component";
