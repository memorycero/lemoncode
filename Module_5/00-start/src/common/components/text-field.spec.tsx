import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { TextField } from './text-field';

describe('text-field component specs', () => {
    it('should display the component empty when it mounts', () => {
        //Arrange
        const props = {
            name: 'name',
            label: 'label',
            value: 'value',
            onChange: jest.fn()
        };
        //Act
        const { asFragment } = render(<TextField {...props} />);

        //Assert
        expect(asFragment()).toMatchSnapshot();
    });

    it('should call onChange property when the text change in the text field component', () => {
        // Arrange
        const props = {
            name: 'name',
            label: 'label',
            value: 'value',
            onChange: jest.fn()
        };

        // Act
        const { getByDisplayValue } = render(<TextField {...props} />);
        fireEvent.change(getByDisplayValue(props.value), { target: { value: 'new value' } });

        // Assert
        expect(props.onChange).toHaveBeenCalled();
        expect(props.onChange).toHaveBeenCalledTimes(1);

    });

    it('should call onBlur method when the component loose focus', () => {
        // Arrange
        const props = {
            name: 'name',
            label: 'label',
            value: 'value',
            onChange: jest.fn(),
            onBlur: jest.fn()
        };

        // Act
        const { getByDisplayValue } = render(<TextField {...props} />);
        fireEvent.blur(getByDisplayValue(props.value));

        // Assert
        expect(props.onBlur).toHaveBeenCalled();
        expect(props.onBlur).toHaveBeenCalledTimes(1);

    });

});
