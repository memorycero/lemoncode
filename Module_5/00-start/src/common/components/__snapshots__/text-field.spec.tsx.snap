// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text-field component specs should display the component empty when it mounts 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
    >
      label
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        type="text"
        value="value"
      />
    </div>
  </div>
</DocumentFragment>
`;
